math/fibonacci.c

```diff
@@ -56,14 +56,19 @@ static void test()
 int main()
 {
     int number;
+    char buffer[8];
 
     // Performing the test
     test();
     printf("Tests passed...\n");
 
     // Getting n
     printf("Enter n to find nth fibonacci element: ");
-    scanf("%d", &number);
+    scanf("%2[0-9]s", buffer);
+
+    number = atoi(buffer);
```

Comment:
`atoi()` is not a part of the C standard, use `strtol()` or `sscanf()` instead.

Reply:
How come it is not part of the C standard? It is present in <stdlib.h>, right?

Reply:
> How come it is not part of the C standard? It is present in <stdlib.h>, right?

That's another thing your teacher didn't told you. You most likely know that there's a lot of different C compilers out there(gcc, MSVC, clang, just to name a few), and in order for a compiler to be a C compiler, it only needs to include all features specified in the C standard, how these features are implemented and what additional features to include are entirely up the compiler's developer's discretion. `atoi()` is *not* a part of the ANSI standard, but it is a commonly included additional feature, hence it is present in *your copy* of the `stdlib.h`

Reply:
Can you please share any resources to check whether a function belongs to the C standard?

Reply:
> Can you please share any resources to check whether a function belongs to the C standard?

Unfortunately I am not aware of any resources of this kind, but generally speaking, searching "[function name] best practice" should yield some useful results

Reply:
Thanks